### ISSUE TITLE
Proper systemLocale for android and iOS

### ISSUE THREAD

**Issue by zeromod** [comahe-de/i18n4k#72]
Android:
Currently `systemLocale` for android is using the jvm variant of `Locale.getDefault`. But this has been deprecated from Android N. [Currently the proper way to get the device locale](https://stackoverflow.com/a/49256800) (without Android Context) is 
```
ConfigurationCompat.getLocales(Resources.getSystem().configuration).get(0)
```

iOS:
`NSLocale.currentLocale` has been deprecated since iOS 11. Replacement for this API is `NSLocale.autoupdatingCurrentLocale` ([source](https://developer.apple.com/documentation/foundation/nslocale/1414388-autoupdatingcurrentlocale))

**Comment by comahe-de** [comment]
@zeromod Thank you for your hints.

The change for iOS was easy.

But I have tried several hours to add a `androidTarget` to the project. Gradle always told me that:

```
Build file 'C:\home\git\github\comahe-de\i18n4k\i18n4k-core\build.gradle.kts' line: 5

An exception occurred applying plugin request [id: 'com.android.library']
> Failed to apply plugin 'com.android.library'.
   > Can't infer current AndroidGradlePluginVersion: Is the Android plugin applied?
```

See commit [18ecfde](https://github.com/comahe-de/i18n4k/commit/18ecfdeccdfacec2d4ba8f3df01d90c3431e4744)

Pipeline: https://github.com/comahe-de/i18n4k/actions/runs/12552537878/job/34998667786#step:7:265

Does anyone has any idea what the problem is?

**Comment by zeromod** [comment]
Yeah sure, I can create a PR for adding `androidTarget` and the change needed for `systemLocale`.

**Comment by zeromod** [comment]
@comahe-de I figured out the reason why it was failing.
The Kotlin plugin in [i18n4k.kmp-build](https://github.com/comahe-de/i18n4k/blob/6644f5b3f87d7bdec69ec816aa075b2e37363cdd/buildSrc/src/main/kotlin/i18n4k.kmp-build.gradle.kts#L4) expects android library plugin in there. Because the order of plugins matter in this case. 
So we need to either 
1. Add the android library plugin in `i18n4k.kmp-build`, this will affect other modules as well.
2. Create a separate variant of `i18n4k.kmp-build` with android library plugin.

**Comment by zeromod** [comment]
@comahe-de I have a PR #73 towards your branch. I haven't done much testing yet, hope this helps.

**Comment by comahe-de** [comment]
@zeromod Thank you for the PR. That was the missing piece for Gradle. I will test it and create a new version in the next days.

P.S: I wish you a happy new year!

**Comment by zeromod** [comment]
Happy new year ✨🎉
